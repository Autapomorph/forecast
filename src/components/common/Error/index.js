import * as React from 'react';

import Message from '../Message';

const Error = ({ children }) => <Message>{children}</Message>;

export default Error;
