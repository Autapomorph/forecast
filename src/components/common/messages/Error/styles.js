import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
  font-size: 1.3rem;
`;
