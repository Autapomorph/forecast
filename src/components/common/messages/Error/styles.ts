import styled from 'styled-components/macro';

import Message from 'components/common/messages/Message';

export const StyledErrorMessage = styled(Message)`
  background: var(--bg-error);
  color: var(--error-text-color);
`;
