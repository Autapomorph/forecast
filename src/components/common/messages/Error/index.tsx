import React from 'react';

import { StyledErrorMessage } from './styles';

const Error: React.FC = ({ children, ...props }): React.ReactElement => (
  <StyledErrorMessage {...props}>{children}</StyledErrorMessage>
);

export default Error;
