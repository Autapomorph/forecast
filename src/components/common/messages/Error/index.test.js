import Error from '.';

describe('Error tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<Error />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render without children', () => {
    const wrapper = shallow(<Error />);

    expect(wrapper.children()).toHaveLength(0);
  });

  it('should render single child', () => {
    const child = <div>child</div>;
    const wrapper = shallow(<Error>{child}</Error>);

    expect(wrapper.children()).toEqual(shallow(child));
  });

  it('should render children', () => {
    const { Fragment } = React;

    const children = (
      <Fragment>
        <div>child</div>
        <div>another child</div>
      </Fragment>
    );
    const wrapper = shallow(<Error>{children}</Error>);

    expect(wrapper.children()).toHaveLength(2);
  });
});
