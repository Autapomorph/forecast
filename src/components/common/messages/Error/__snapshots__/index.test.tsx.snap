// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error tests renders properly 1`] = `<styles__StyledErrorMessage />`;
